src/RequestParser.cpp

```diff
@@ -734,3 +811,91 @@ void RequestParser::checkTransferEncoding(HTTPRequest& request)
 		}
 	}
 }
+
+/**
+ * @brief Checks if the HTTP request body is complete based on headers.
+ *
+ * This function checks whether the HTTP request body is fully received
+ * by examining the `Content-Length` and `Transfer-Encoding` headers.
+ *
+ * - If the `Content-Length` header is present and `Transfer-Encoding` is not,
+ *   the function compares the actual body size with the specified content length.
+ * - If the body size exceeds the specified `Content-Length`, it sets the
+ *   HTTP status to `400 Bad Request` and throws an exception.
+ * - If the `Transfer-Encoding` header is present, it checks for the presence
+ *   of the termination sequence (`0\r\n\r\n`) to determine completeness.
+ *
+ * @param bodyString The actual body of the HTTP request received so far.
+ * @param request The HTTP request object containing headers and status.
+ *
+ * @return true If the body is complete.
+ * @return false If the body is not yet complete.
+ *
+ * @throws std::runtime_error If the body size exceeds the specified `Content-Length`.
+ */
+bool RequestParser::checkForCompleteBody(const std::string& bodyString, HTTPRequest& request)
```

Comment:
I tried to find out what the behaviour of our server should be, when the content-length is bigger than the body size. 

My search was not unfortunately not successful (looked into RFCs, in the doc of nginx etc.). 

I see that you do not set any specific status code for this case and just normally process such a request. 
Is that on purpose?

Reply:
If the received body is less than the indicated Content-Length than this function will never return true, therefore the parsing of the body will never be triggered. Eventually the connection will simply timeout as the server expects more data, but receives no more.

Reply:
Thanks for the clarification. Makes senses.